src/Files.App/MainWindow.xaml.cs

```diff
@@ -339,5 +342,28 @@ x.tabItem.NavigationParameter.NavigationParameter is PaneNavigationArguments pan
 				}
 			}
 		}
+
+		public bool SetCanWindowToFront(bool canWindowToFront)
+		{
+			if (CanWindowToFront != canWindowToFront)
+			{
+				CanWindowToFront = canWindowToFront;
+				return true;
+			}
+			return false;
```

Comment:
This does not guarantee atomic operation when multiple threads handle it at the same time. It is better to use `Interlocked` class or `lock` statement.